### PR TITLE
fix(diff): handle edge case diffing empty against non-empty

### PR DESCRIPTION
**Pull request by frrist**
- and add tests

### CODE REVIEW

File: diff.go [hunk 1]
```diff
@@ -67,6 +67,13 @@ func Diff(ctx context.Context, prevBs, curBs cbor.IpldStore, prev, cur cid.Cid,
 		height:   curAmt.height,
 	}
 
+	// edge case of diffing an empty AMT against non-empty
+	if prevAmt.count == 0 && curAmt.count != 0 {
+		return addAll(ctx, curCtx, curAmt.node,0)
```
Comment (rvagg) [inline diff.go:72]: unspaced `,`, needs a `go fmt`

**Review by rvagg — APPROVED** [review]
